### PR TITLE
Add extra build parameters and flags

### LINKED ISSUE #25: Feature to add additional xcconfig

### PR DESCRIPTION
**Pull request by giginet**
closes #25

Added a feature to pass extra build parameters and flags.

### CODE REVIEW

File: Sources/ScipioKit/Producer/PIF/XCBuildClient.swift [hunk 1]
```diff
@@ -7,17 +7,20 @@ struct XCBuildClient {
...
     private let executor: any Executor
+    private let buildExecutor: any Executor
 
     init(
         package: DescriptionPackage,
         buildProduct: BuildProduct,
         configuration: BuildConfiguration,
-        executor: any Executor = ProcessExecutor(decoder: XCBuildOutputDecoder())
+        executor: any Executor = ProcessExecutor(decoder: StandardOutputDecoder()),
+        xcBuildExecutor: any Executor = ProcessExecutor(decoder: XCBuildOutputDecoder())
```
Comment (giginet) [inline Sources/ScipioKit/Producer/PIF/XCBuildClient.swift:17]: `xcbuild build` outputs JSON format. So it needs a special decoder.

File: Sources/ScipioKit/Producer/PIF/PIFCompiler.swift [hunk 2]
```diff
@@ -115,6 +115,7 @@ struct PIFCompiler: Compiler {
             toolchain: toolchain,
             destinationTriple: toolchain.triple,
             flags: .init(),
+            enableParseableModuleInterfaces: true,
```
Comment (giginet) [inline Sources/ScipioKit/Producer/PIF/PIFCompiler.swift:118]: This value is not used in Scipio. So it doesn't affect anything.
But `true` is the right state.

This option means whether emitting swift modules or not.

File: Sources/ScipioKit/Producer/PIF/BuildParametersGenerator.swift [hunk 3]
```diff
@@ -50,33 +52,35 @@ struct BuildParametersGenerator {
...
 
-        settings["FRAMEWORK_SEARCH_PATHS"] = ["$(inherited)", "$(BUILT_PRODUCTS_DIR)/PackageFrameworks"].joined(separator: " ")
+        settings["FRAMEWORK_SEARCH_PATHS"] = expandFlags(
+            "$(BUILT_PRODUCTS_DIR)/PackageFrameworks"
+        )
 
-        // Optionally also set the list of architectures to build for.
-        if let architectures = buildParameters.architectures, !architectures.isEmpty {
-            settings["ARCHS"] = architectures.joined(separator: " ")
-        }
```
Comment (freddi-kit) [inline Sources/ScipioKit/Producer/PIF/BuildParametersGenerator.swift:79]: We no longer use it, right?

File: Sources/ScipioKit/Runner.swift [hunk 4]
```diff
@@ -275,7 +288,9 @@ extension Runner.Options.BuildOptions {
             platforms: fetch(\.platforms, by: \.platforms),
             isSimulatorSupported: fetch(\.isSimulatorSupported, by: \.isSimulatorSupported),
             isDebugSymbolsEmbedded: fetch(\.isDebugSymbolsEmbedded, by: \.isDebugSymbolsEmbedded),
-            frameworkType: fetch(\.frameworkType, by: \.frameworkType)
+            frameworkType: fetch(\.frameworkType, by: \.frameworkType),
+            extraFlags: overridingOptions.extraFlags ?? extraFlags,
+            extraBuildParameters: overridingOptions.extraBuildParameters ?? extraBuildParameters
```
Comment (freddi-kit) [inline Sources/ScipioKit/Runner.swift:293]: It looks all `extraBuildParameters` will be ignored if we set `overridingOptions.extraBuildParameters`, is it okay? We don't combine them, right?

Comment (giginet) [reply Sources/ScipioKit/Runner.swift:293]: Good point!

Let's combine two settings

**Review by freddi-kit — APPROVED** [review]

**Pull request merged**